src/site/content/en/blog/text-fragments/index.md

```diff
@@ -0,0 +1,422 @@
+</figure>
+
+### The fragment directive
+
+So far I have only quickly glanced over one important addition without actually explaining it:
+the fragment directive `:~:`.
+To avoid compatibility issues with the usage of existing URL element fragments as shown above,
+the [Text Fragments specification](https://wicg.github.io/ScrollToTextFragment/)
+introduces the fragment directive.
+The fragment directive is a portion of the URL fragment delimited by the code sequence `:~:`.
+It is reserved for user agent instructions, such as `text=`,
+and is stripped from the URL during loading so that author scripts cannot directly interact with it.
+User agent instructions are also called *directives*.
+In the concrete case, `text=` is therefore called a *text directive*.
+
+### Feature detectability
+
+Support for Text Fragments is feature-detectable
+by checking for the existence of the `fragmentDirective` in the `Location` interface.
+Unlike the other properties like `hash` that I have mentioned above,
+`fragmentDirective` does not return any value and is read-only.
+The fragment directive is a mechanism for URLs to specify instructions
+directed to the browser rather than the document.
+It is meant to avoid direct interaction with author script,
+so that future user agent instructions can be added without fear
+of introducing breaking changes to existing content.
+One potential example of such future additions could be translation hints.
+
+```js
+if ('fragmentDirective' in window.location) {
+  // Text Fragments is supported.
+}
+```
+
+### Polyfillability
+
+The Text Fragments feature can be polyfilled to some extent.
+There is early [work in progress](https://github.com/tomayac/text-fragments-polyfill/)
+to create an extension for browsers that do not support Text Fragments natively
+where the functionality is implemented in JavaScript.
+
+### Security
+
+Text fragment directives are invoked only on full (non-same-page) navigations
+that are the result of a
+[user activation](https://html.spec.whatwg.org/multipage/interaction.html#tracking-user-activation).
+Additionally, navigations originating from a different origin than the destination
+will require the navigation to take place in a
+[`noopener`](https://html.spec.whatwg.org/multipage/links.html#link-type-noopener) context,
+such that the destination page is known to be sufficiently isolated.
+Text fragment directives are only applied to the main frame.
+This means that text will not be searched inside iframes,
+and iframe navigation will not invoke a text fragment.
+
+### Privacy
+
+It is important that implementations of the Text Fragments specification
+do not leak whether a text fragment was found on a page or not.
+While element fragments are fully under the control of the original page author
+(remember how there was no way to link to the section
+I wanted to link to before in my running example),
```

Comment:
What?

Reply:
There was no element identifier `id="something"` in the blog post, so I couldn't link.

Reply:
Then that's how you should say it. It's easier to understand. Also I'd put this after the current sentence.

Reply:
I have rephrased.